### ISSUE TITLE
Very odd promise rejection crash

### ISSUE THREAD

**Issue by a-s-o** [petkaantonov/bluebird#199]
Hello,

I am using bluebird and getting a really odd crash when I try to reject a promise, but only in one function. Not sure what I could be doing wrong as I have used same methods in many other places without issue. Here is the stack trace for the crash

```
E:\app\server\node_modules\bluebird\js\main\async.js:93
                throw res.e;
                         ^
Error: Invalid put request
    at app.backbone.model.extend.put (E:\app\server\initializers\databases.js:179:23)
    at Object.exports.putDocument.run (E:\app\server\actions\crud.js:59:11)
    at Object.app.faye.addExtension.incoming (E:\app\server\initializers\faye.js:87:23)
    at pipe (E:\app\server\node_modules\faye\node\faye-node.js:901:43)
    at Object.Faye.Extensible.pipeThroughExtensions (E:\app\server\node_modules\faye\node\faye-node.js:904:5)
    at Object.Faye.Server.Faye.Class.process (E:\app\server\node_modules\faye\node\faye-node.js:1964:12)
    at Object.Faye.NodeAdapter.Faye.Class._callWithParams (E:\app\server\node_modules\faye\node\faye-node.js:2607:20)
    at E:\app\server\node_modules\faye\node\faye-node.js:2584:14
    at ConcatStream.<anonymous> (E:\app\server\node_modules\faye\node_modules\concat-stream\index.js:32:43)
    at ConcatStream.EventEmitter.emit (events.js:117:20)
6 May 13:17:38 - [nodemon] app crashed - waiting for file changes before starting...
```

I have tried throwing the error in the following different ways:

Promise.try() -> throw
new Promise() -> throw
new Promise(function(resolve, reject)) -> reject
Promise.defer() -> deferred.reject
Promise.reject()

They all lead to this crash. I have tried throwing a string, object, new Error() and still the same result. I tried searching the issues and google but did not find an answer so far, so please let me know if there is a certain direction to be checked.

I am returning the resulting promise. I am able to `promise.catch()` while in the same function but once it is returned, it seems to crash.

Thanks, and thanks for a great library. Appreciate any help.
